meteor/server/api/peripheralDevice.ts

```diff
@@ -256,12 +257,19 @@ export namespace ServerPeripheralDeviceAPI {
 							PlayoutLockFunctionPriority.CALLBACK_PLAYOUT,
 							async () => {
 								const rundownIDs = Rundowns.find({ playlistId }).map((r) => r._id)
+								const partInstanceIDs = [
```

Comment:
We shouldn't rely on `activePlaylist` inside the lock, as other playout operations could be done between loading that playlist and aquiring the lock causing us to use stale data.

Reply:
Aren't we acquiring a lock & cache for that specific playlist anyway implying that if the playlist changes we are working with stale/wrong data anyway?

Reply:
I dont follow that

Reply:
In line 256 the playlistId is used to load the cache, reusing that id inside the function that has the cache seems safe.

Reply:
using the id to load the cache is different to relying on properties on the playlist to not have changed

but I completely missed that this is inside of a `runStudioOperationWithCache` lock, so ignore me. that guarantees that nothing else will be done in the middle (I know, the locks in sofie at the moment are a bit confusing..)